sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/BulkWriter.scala

```diff
@@ -662,7 +671,10 @@ private class BulkWriter(container: CosmosAsyncContainer,
 
       throwIfProgressStaled(
         "Semaphore acquisition",
+        // TODO @fabianm BEFORE MERGE use empty set instead
         activeOperationsSnapshot,
+        activeOperationsSnapshot,
```

Comment:
should here be pendingBulkWriteRetires snapshot? same for the readMany

Reply:
Good catch - but I should suppress the new retry mechanism here completely - while we are still scheduling writes this won't be safe

Reply:
Fixed